buildSrc/src/main/kotlin/net/skyscanner/backpack/tokens/BpkColors.kt

```diff
@@ -46,58 +49,78 @@ object BpkColor {
     override fun invoke(source: Map<String, Any>): BpkColors =
       parseColors(source, resolveReferences = false) {
         it.name == it.defaultReference && !it.isMarcomms()
-      }
+      }.toBpkColors()
   }
 
   object Semantic : BpkParser<Map<String, Any>, BpkColors> {
 
     override fun invoke(source: Map<String, Any>): BpkColors =
-      parseColors(source, resolveReferences = true) {
-        it.name != it.defaultReference && !it.hasSemanticSuffix() && !it.isMarcomms()
-      }
+      parseColors(source, resolveReferences = true) { it.isSemanticColor() && !it.isPrivate }.toBpkColors()
+  }
 
-    @OptIn(ExperimentalStdlibApi::class)
-    private fun BpkColorModel.hasSemanticSuffix(): Boolean {
-      val name = name.lowercase()
-      return semanticSuffixes.any { name.endsWith("_$it") && !name.endsWith("_on_$it") }
-    }
+  object Internal : BpkParser<Map<String, Any>, Map<String, BpkColors>> {
```

Comment:
there's now a third type of colours for internal tokens where we want dynamic tokens, but don't care about theming support